fdb-extensions/src/main/java/com/apple/foundationdb/async/RTree.java

```diff
@@ -163,49 +163,130 @@ public class RTree {
      *     respect to each other. Example: {@code MIN_M = 25}, {@code MAX_M = 32}, {@code S = 2}, two nodes at
      *     already at maximum capacity containing a combined total of 64 children when a new child is inserted.
      *     We split the two nodes into three as indicated by {@code S = 2}. We have 65 children but there is no way
-     *     of distributing them among three nodes such that none of them underflows.</li>
+     *     of distributing them among three nodes such that none of them underflows. This constraint can be
+     *     formulated as {@code S * MAX_M / (S + 1) >= MIN_M}.</li>
      *     <li>When fusing {@code S + 1} to {@code S} nodes, we re-distribute the children of {@code S + 1} nodes
      *     into {@code S + 1} nodes which may cause an overflow if {@code S} and {@code M} are not set carefully with
      *     respect to each other. Example: {@code MIN_M = 25}, {@code MAX_M = 32}, {@code S = 2}, three nodes at
      *     already at minimum capacity containing a combined total of 75 children when a child is deleted.
      *     We fuse the three nodes into two as indicated by {@code S = 2}. We have 75 children but there is no way
-     *     of distributing them among two nodes such that none of them overflows.</li>
+     *     of distributing them among two nodes such that none of them overflows. This constraint can be formulated as
+     *     {@code (S + 1) * MIN_M / S <= MAX_M}.</li>
      * </ol>
+     * Both constraints are in fact the same constraint and can be written as {@code MAX_M / MIN_M >= (S + 1) / S}.
      */
     public static final int DEFAULT_S = 2;
 
+    /**
+     * Default storage layout. Can be either {@code BY_SLOT} or {@code BY_NODE}. {@code BY_SLOT} encodes all information
+     * pertaining to a {@link NodeSlot} as one key/value pair in the database, {@code BY_NODE} encodes all information
```

Comment:
```suggestion
     * pertaining to a {@link NodeSlot} as one key/value pair in the database; {@code BY_NODE} encodes all information
```

It could be a "." or a comma and a preposition. Just needs something more intense than a comma

Reply:
done.